picard/ui/itemviews.py

```diff
@@ -337,64 +451,47 @@ def select_object(self, obj):
                 break
 
 
-def paint_column_icon(painter, rect, icon):
+def paint_column_icon(painter, rect, column):
```

Comment:
The history of this function is that it originally was used for painting both the header and the content of the cells, but it was using different icons (gray in the header, gray or colored in the columns). This was later changed to use `TreeItem.setIcon()` in the content cells for performance reason (even though it causes a slight misalignment, but it is much faster than a custom paint).

Anyway, my suggestion would be to move the `paint_column_icon` to `IconColumn.paint_header(painter, rect)`, as it now is tightly couple to the IconColumn class anyway and also is only about painting the header. Maybe also rename `IconColumn.icon` to `IconColumn.header_icon`.

Reply:
Done in https://github.com/metabrainz/picard/pull/2454/commits/5fb55d30e0fccafbfc58f11dc926a0ae9ed35e65 and https://github.com/metabrainz/picard/pull/2454/commits/bc99b478b8a2715a1db50d78b810a864108c4df9